### PR TITLE
Documentation 3xx redirects can't send headers

### PR DESCRIPTION
**Pull request by MichaelWest22**
## Description
Added some documentation for how response headers are not received or processed when 3xx response codes are returned.

Corresponding issue:
#2879 

## Checklist

* [x] I have read the contribution guidelines
* [x] I have targeted this PR against the correct branch (`master` for website changes, `dev` for
  source changes)
* [x] This is either a bugfix, a documentation update, or a new feature that has been explicitly
  approved via an issue
* [ ] I ran the test suite locally (`npm run test`) and verified that it succeeded

### CODE REVIEW

File: www/content/docs.md [hunk 1]
```diff
@@ -1028,13 +1028,13 @@ htmx includes a number of useful headers in requests:
 | `HX-Trigger-Name` | the `name` of the triggered element if it exists
 | `HX-Trigger` | the `id` of the triggered element if it exists
 
-### Response Headers
+### Response Headers {#response_headers}
```
Comment (Telroshan) [inline www/content/docs.md:1031]: This seems to break the default anchor that was `response-headers` and referenced at 2 other places in this file.
Ran `zola serve` and got 2 errors, I'd say either replace this by `response-headers` or replace the 2 existing usages by `response_headers`.

**Review by Telroshan — CHANGES_REQUESTED** [review]
Great addition for a very common gotcha!

File: www/content/docs.md (hunk 1 above)
Comment (MichaelWest22) [reply www/content/docs.md]: Yeah sorry I missed those two existing links.  should be fixed now

**Review by Telroshan — APPROVED** [review]

**Comment by Telroshan** [comment]
Perfect thanks!

**Pull request merged**